site/en/r2/tutorials/beginner/tf2_overview.ipynb

```diff
@@ -312,8 +334,8 @@
         "    logits = model(images, training=True)\n",
         "    loss = compute_loss(logits, labels)\n",
         "    accuracy = compute_accuracy(logits, labels)\n",
-        "  grads = tape.gradient(loss, model.trainable_variables)\n",
-        "  optimizer.apply_gradients(zip(grads, model.trainable_variables))\n",
+        "  grads = tape.gradient(loss, model.variables)\n",
+        "  optimizer.apply_gradients(zip(grads, model.variables))\n",
```

Comment:
This needs to be `model.trainable_variables`
You might need to rebase with master